examples/apps/color_finder.py

```diff
@@ -0,0 +1,475 @@
+            self.color_to_find_index = 0
+        self.color_to_find = self.possible_colors_to_find[self.color_to_find_index]
+        self.color_selector_cube.set_lights(map_color_to_light[self.color_to_find])
+
+    def update_pixel_matrix(self,downsized_image):
+        ''' Uses the getpixel method from the Pillow library to get the RGB values
+            of all the pixels in the low-res approximation of Cozmo's camera view.
+        '''
+        for i in range(self.grid_width):
+            for j in range(self.grid_height):
+                r,g,b = downsized_image.getpixel((i,j))
+                self.pixel_matrix[i][j] = self.approximate_color_of_pixel(r,g,b)
+        self.fill_in_gaps_of_pixel_matrix()
+
+    def approximate_color_of_pixel(self,r,g,b):
+        ''' Returns the approximate color of this pixel from color_dict.'''
+        min_distance = sys.maxsize
+        closest_color = ''
+        for color_name, color_values in color_dict.items():
+            d = color_distance((r,g,b),color_values)
+            if d < min_distance:
+                min_distance = d
+                closest_color = color_name
+        return closest_color
+
+    def fill_in_gaps_of_pixel_matrix(self):
+        ''' Fills in squares in self.pixel_matrix that meet the condition
+            in self.pixel_is_surrounded_by_color
+        '''
+        for i in range(1,self.grid_width-1):
+            for j in range(1,self.grid_height-1):
+                color = self.pixel_is_surrounded_by_color(i,j)
+                if color:
+                    self.pixel_matrix[i][j]=color
+
+    def pixel_is_surrounded_by_color(self,i,j):
+        ''' If a pixel is surrounded by three pixels of the same color, returns that color. Otherwise, returns None.'''
+        if self.pixel_matrix[i-1][j]==self.pixel_matrix[i][j-1] and self.pixel_matrix[i-1][j]==self.pixel_matrix[i+1][j]:
+            return self.pixel_matrix[i-1][j]
+        if self.pixel_matrix[i-1][j]==self.pixel_matrix[i][j-1] and self.pixel_matrix[i-1][j]==self.pixel_matrix[i][j+1]:
+            return self.pixel_matrix[i-1][j]
+        if self.pixel_matrix[i-1][j]==self.pixel_matrix[i+1][j] and self.pixel_matrix[i-1][j]==self.pixel_matrix[i][j+1]:
+            return self.pixel_matrix[i-1][j]
+        if self.pixel_matrix[i+1][j]==self.pixel_matrix[i][j-1] and self.pixel_matrix[i+1][j]==self.pixel_matrix[i][j+1]:
+            return self.pixel_matrix[i+1][j]
+        return None 
+
+    async def on_new_camera_image(self, evt, **kwargs):
+        ''' Processes the blobs in Cozmo's view.
+            Triggered by EvtOnNewCameraImage ~15 times a second.
+        '''
+        downsized_image = self.get_low_res_view()
+        self.update_pixel_matrix(downsized_image)
+        blob_detector = BlobDetector(self.pixel_matrix,self.possible_colors_to_find)
+        blob_info = blob_detector.get_center_of_blob_with_color(self.color_to_find)
+        if blob_info:
+            if self.state=='look_around':
```

Comment:
instead of using strings for each state name here (which risk a typo in one place) add constants so e.g:
`LOOK_AROUND_STATE = 'look_around'`